rest-service/manager_rest/storage/resource_models.py

```diff
@@ -656,18 +661,123 @@ def _get_identifier_dict(self):
         id_dict['status'] = self.status
         return id_dict
 
-    def set_deployment(self, deployment, blueprint_id=None):
-        self._set_parent(deployment)
-        self.deployment = deployment
-        current_blueprint_id = blueprint_id or deployment.blueprint_id
-        self.blueprint_id = current_blueprint_id
-
     @classproperty
     def resource_fields(cls):
         fields = super(Execution, cls).resource_fields
         fields.pop('token')
         return fields
 
+    @validates('deployment')
+    def _set_deployment(self, key, deployment):
+        self._set_parent(deployment)
+        if self.blueprint_id is None:
```

Comment:
Nitpick: Any reason not to write this as: self.blueprint_id = self.blueprint_id or deployment.blueprint_id?

Reply:
not really, but i think i prefer `is None` wherever possible. You know, performance!